pkgdb/src/buildenv/realise.cc

```diff
@@ -373,6 +378,69 @@ createFloxEnv( nix::EvalState &     state,
 }
 
 
+nix::StorePath
+createContainerBuilder( nix::EvalState & state,
+                        nix::StorePath   environmentStorePath,
+                        const System &   system )
+{
+
+  static const std::string refOrRev
+    = nix::getEnv( "_PKGDB_GA_REGISTRY_REF_OR_REV" )
+        .value_or( "release-23.05" );
+  static const nix::FlakeRef nixpkgsRef
+    = nix::parseFlakeRef( "github:NixOS/nixpkgs/" + refOrRev );
```

Comment:
I dont know if we have any other nixpkgs reference that we already bundle...
Also this is code duplication with https://github.com/flox/flox/blob/f21591738cbbe337d8dc41591a78371d8dc6aad4/pkgdb/src/registry.cc#L379-L384

Reply:
We could do something like https://github.com/flox/flox/blob/f9c580546cc617fa20e8106ea4e655dbfc60b45f/assets/mkEnv/env-from-lockfile.nix#L8. It's a bit weird to make the builder depend on the nixpkgs pinned in the environment, but I think that's better than downloading another nixpkgs

Reply:
I agree on the weird part.. since there may be changes reg the `dockerTools` between nixpkgs releases that we can't account for.

TBF most of `dockerTools` changes fairly infrequently: https://github.com/NixOS/nixpkgs/blame/master/pkgs/build-support/docker/default.nix

Still the lockfile will have more than one nixpkgs input eventually -- how to choose?
It just becomes a bit worse to debug - i'd rather stick to a shipped nixpkgs (either shipped, bigger image :( or fetched lazily as we do now) would just be nice to have that specified in a more obvious place rather than in the depths of two unrelated `.cc` files...

Reply:
I'm reading the nixpkgs url from **this flake's** lockfile now so at least its bumped as we bump the nixpkgs to build flox/pkgdb with.
this also ensures that we reuse parts of the closure of flox/pkgdb where applicable